app/utils/CheckYourAnswersHelper.scala

```diff
@@ -25,25 +25,21 @@ import play.api.i18n.Messages
 import play.twirl.api.{Html, HtmlFormat}
 import viewmodels.AnswerRow
 import CheckYourAnswersHelper._
+import models.TypeOfParticipant.SomethingElse
 
 class CheckYourAnswersHelper(userAnswers: UserAnswers)(implicit messages: Messages) {
 
   def typeOfParticipant: Option[AnswerRow] = userAnswers.get(TypeOfParticipantPage) map {
     x =>
       AnswerRow(
         HtmlFormat.escape(messages("typeOfParticipant.checkYourAnswersLabel")),
-        Html(x.map(value => HtmlFormat.escape(messages(s"typeOfParticipant.$value")).toString).mkString(",<br>")),
-        routes.TypeOfParticipantController.onPageLoad(CheckMode).url
-      )
-  }
-
-  def typeOfParticipantDetail: Option[AnswerRow] = userAnswers.get(TypeOfParticipantDetailPage) map {
-    x =>
-      AnswerRow(
-        HtmlFormat.escape(messages("typeOfParticipantDetail.checkYourAnswersLabel")),
-        HtmlFormat.escape(x),
-        routes.TypeOfParticipantDetailController.onPageLoad(CheckMode).url
-      )
+    Html(x.map(value => if(value == SomethingElse) {
+      userAnswers.get(TypeOfParticipantDetailPage).getOrElse("")
+    } else{
+      HtmlFormat.escape(messages(s"typeOfParticipant.$value")).toString
+    }).mkString(",<br>")),
+    routes.TypeOfParticipantController.onPageLoad(CheckMode).url
+    )
```

Comment:
@Dandyboyo2 

The following code will prevent the ugly "," appearing at the end of the list when the user selects another type of AMP however; uses the back button to check your answers before submitting the detail test. 

```
      val rowVal = Html(x.map(value => if(value == SomethingElse) {
        userAnswers.get(TypeOfParticipantDetailPage).getOrElse("")
      } else{
        HtmlFormat.escape(messages(s"typeOfParticipant.$value")).toString
      }).mkString(",<br>"))

      val formattedRowVal = if (rowVal.toString().endsWith(",<br>")) {
        Html(rowVal.toString().substring(0, rowVal.toString().length - 5))
      } else {
        rowVal
      }

      AnswerRow(
        HtmlFormat.escape(messages("typeOfParticipant.checkYourAnswersLabel")),
        formattedRowVal,
        routes.TypeOfParticipantController.onPageLoad(CheckMode).url
      )
```

Reply:
This could be further refactored into its own method;

`  private def formatRowValTypeOfParticipant(x: Seq[TypeOfParticipant]): Html = {
    val rowVal = Html(x.map(value => if(value == SomethingElse) {
      userAnswers.get(TypeOfParticipantDetailPage).getOrElse("")
    } else{
      HtmlFormat.escape(messages(s"typeOfParticipant.$value")).toString
    }).mkString(",<br>"))

    if (rowVal.toString().endsWith(",<br>")) {
      Html(rowVal.toString().substring(0, rowVal.toString().length - 5))
    } else {
      rowVal
    }
  }

  def typeOfParticipant: Option[AnswerRow] = userAnswers.get(TypeOfParticipantPage) map {
    x: Seq[TypeOfParticipant] =>
      AnswerRow(
        HtmlFormat.escape(messages("typeOfParticipant.checkYourAnswersLabel")),
        formatRowValTypeOfParticipant(x),
        routes.TypeOfParticipantController.onPageLoad(CheckMode).url
      )
  }`